src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,196 @@
+
+type Props = {
+  setShowFilters: (arg: boolean) => void;
+};
+
+export const PeopleTable: React.FC<Props> = ({ setShowFilters }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('century');
+
+  const getSortParams = (value: string) => ({
+    sort: sort === value && order === 'desc' ? null : value,
+    order: sort === value && order === null ? 'desc' : null,
+  });
+
+  const sortFilters = ['Name', 'Sex', 'Born', 'Died'];
+
+  const handleError = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => setErrorMessage(''), 3000);
+  };
+
+  useEffect(() => {
+    const getPeopleFromServer = async () => {
+      try {
+        setLoading(true);
+        const peopleFromServer = await getPeople();
+
+        const peopleWithParents = peopleFromServer.map(person => ({
+          ...person,
+          mother: peopleFromServer.find(
+            parent => parent.name === person.motherName,
+          ),
+          father: peopleFromServer.find(
+            parent => parent.name === person.fatherName,
+          ),
+        }));
+
+        setPeople(peopleWithParents);
+        setShowFilters(true);
+      } catch {
+        handleError('Something went wrong');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getPeopleFromServer();
+  }, []);
+
+  function sortPeople(peopleToSort: Person[]): Person[] {
+    const peopleSorted = [...peopleToSort];
+
+    const sorting = (a: Person, b: Person): number => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return a[sort].localeCompare(b[sort]);
+
+        case 'born':
+        case 'died':
+          return +a[sort] - +b[sort];
+
+        default:
+          return 0;
+      }
+    };
+
+    return order === 'desc'
+      ? peopleSorted.sort((a, b) => sorting(b, a))
+      : peopleSorted.sort((a, b) => sorting(a, b));
+  }
+
+  function filterByQuery(person: Person) {
+    if (!query) {
+      return true;
+    }
+
+    const normalizedQuery = query.toLowerCase();
+
+    return (
+      person.name.toLowerCase().includes(normalizedQuery) ||
+      person.fatherName?.toLowerCase().includes(normalizedQuery) ||
+      person.motherName?.toLowerCase().includes(normalizedQuery)
+    );
+  }
+
+  function filterByCentury(person: Person) {
+    if (centuries.length === 0) {
+      return true;
+    }
+
+    return centuries.includes(Math.ceil(person.born / 100).toString());
+  }
+
+  function filterBySex(person: Person) {
+    if (!sex) {
+      return true;
+    }
+
+    return person.sex === sex;
+  }
+
+  const getPreparedPeople = () => {
```

Comment:
it's better to write it with useMemo:

```suggestion
  const preparedPeople = useMemo(() => {
  // filtering logic
```

but in this case, you have to pass the filter param as an argument to each filter callback and define each callback function outside of the component

Reply:
Fixed.